### PR TITLE
WIP: Updated Dribbler Thermal Model

### PR DESCRIPTION
**Pull request by henryymliu**
### Description

@hannahvsawiuk updated the dribbler motor thermal model, theoretically allowing for much higher dribbler motor speeds.

### Testing Done

Needs IRL robot validation.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by MathewMacDougall — COMMENTED** [review]
LGTM. Going to wait until this is validated on a robot before approving.

File: src/thunderbots/firmware/main/dribbler.c [hunk 1]
```diff
@@ -37,9 +39,9 @@
 #define THERMAL_WARNING_STOP_TEMPERATURE_WINDING (THERMAL_WARNING_START_TEMPERATURE_WINDING - 15.0f) // °C—chead
 #define THERMAL_WARNING_STOP_ENERGY_WINDING ((THERMAL_WARNING_STOP_TEMPERATURE_WINDING - THERMAL_AMBIENT) * THERMAL_CAPACITANCE_WINDING) // joules
 
-#define DRIBBLER_SPEED_BUFFER_ZONE 200 // RPM
+#define DRIBBLER_SPEED_BUFFER_ZONE 250 // RPM
 
-#define MAX_DRIBBLER_CURRENT 5 // Limit the max current the dribbler can draw when stalled
+#define MAX_DRIBBLER_CURRENT 4.0 // Limit the max current the dribbler can draw when stalled
```
Comment (EvanMorcom) [inline src/thunderbots/firmware/main/dribbler.c]: If we are throwing 'f's on the end of all the previous defines, let's also do it here.

File: src/thunderbots/firmware/main/dribbler.c [hunk 2]
```diff
@@ -14,19 +14,21 @@
...
 #define VOLTS_PER_RPM (1.0f / SPEED_CONSTANT) // volts per rpm
 #define VOLTS_PER_RPT (VOLTS_PER_RPM * 60.0f * DRIBBLER_TICK_HZ) // volts per rpt, rpt=revolutions per tick
 #define VOLTS_PER_SPEED_UNIT (VOLTS_PER_RPT / 6.0f) // volts per Hall edge
-#define PHASE_RESISTANCE 0.403f // ohms—EC16 datasheet
+#define PHASE_RESISTANCE 0.512f // ohms—EC16 datasheet
 #define SWITCH_RESISTANCE (0.019f*2.0f) // ohms—AO4882 datasheet
 
-#define THERMAL_TIME_CONSTANT_WINDING 1.97f // seconds—EC16 datasheet
-#define THERMAL_RESISTANCE_WINDING 1.68f // kelvins per Watt—EC16 datasheet (winding to housing)
+#define THERMAL_TIME_CONSTANT_WINDING 2.17f // seconds—EC16 datasheet
```
Comment (EvanMorcom) [inline src/thunderbots/firmware/main/dribbler.c]: Maybe link a datasheet if possible here.

**Review by EvanMorcom — COMMENTED** [review]
:fire:

File: src/thunderbots/firmware/main/dribbler.c (hunk 2 above)
Comment (MathewMacDougall) [reply src/thunderbots/firmware/main/dribbler.c]: @hannahvsawiuk 	for this one

Comment (hannahvsawiuk) [reply src/thunderbots/firmware/main/dribbler.c]: yee, I will grab the link

Comment (hannahvsawiuk) [reply src/thunderbots/firmware/main/dribbler.c]: https://www.maxongroup.com/medias/sys_master/root/8833419509790/19-EN-177.pdf

Comment (henryymliu) [reply src/thunderbots/firmware/main/dribbler.c]: Added datasheet link to top of constants section.

File: src/thunderbots/firmware/main/dribbler.c (hunk 1 above)
Comment (henryymliu) [reply src/thunderbots/firmware/main/dribbler.c]: fixed

**Comment by MathewMacDougall** [comment]
Let's test this Saturday and then I think it's good to go :+1:

**Comment by EvanMorcom** [comment]
@henryymliu @hannahvsawiuk Any progress on this?

**Comment by MathewMacDougall** [comment]
Minor conflict now but the file just needs to be moved up one level. @hannahvsawiuk @henryymliu this was validated Saturday so can be merged?

**Comment by henryymliu** [comment]
Well... there was an interesting issue on mrftest where the speed  maxed out at ~11500 RPM, unsure if that's a firmware bug or some limitation with mrftest or a Hall sensor issue (????).

But it didn't blow up? So it's safe to merge right???!!!1

**Comment by MathewMacDougall** [comment]
Yea the speed limitation is unknown. Probably safe to merge and we can update the models later again when firmware is redone anyway. @hannahvsawiuk your call

**Comment by MathewMacDougall** [comment]
@hannahvsawiuk should we merge this?

**Comment by hannahvsawiuk** [comment]
Yes, it is fine

**Comment by hannahvsawiuk** [comment]
The max is due to the theoretic thermal limits as well as the fact that we run it at a lower nominal voltage so we can only achieve about 87-93% of the max speed.

**Comment by hannahvsawiuk** [comment]
On mobile, but I approve this for merge

**Review by hannahvsawiuk — APPROVED** [review]
Looks good, and has been tested.

**Review by MathewMacDougall — APPROVED** [review]

**Pull request merged**